reeimplemented client data request using observables closes #12

tikisailor opened the pull request:
Now possible to fetch data directly from the get endpoint. If client is down it will timeout and error after 3 seconds

Joelone reviewed (approved):
LGTM
Suggestion: add the reason of re-implement or link it to existed issue

tikisailor commented:
> LGTM Suggestion: add the reason of re-implement or link it to existed issue

Thanks Joel, this change was suggested to avoid front end (or other consumers of this data) having to implement event source.

tikisailor commented:
I added a bugfix I found in the vote logic delete routine and did some cleanup (moved some of the TODO items to github issues, removed old outcommented code, updated api documentation for the reeimplemented client data endpoint)
